packages/namekit-react/src/components/CurrencySymbol.tsx

```diff
@@ -0,0 +1,161 @@
+
+interface CurrencySymbolProps extends React.ComponentProps<React.ElementType> {
+  /**
+   * The `Currency` to display the symbol for.
+   */
+  currency: Currency;
+
+  /**
+   * The size of the `CurrencySymbol` icon.
+   *
+   * Defaults to `CurrencySymbolSize.Small`.
+   * Only applicable when `symbology` is `CurrencySymbology.Icon`.
+   * If you want to set a custom size for other symbologies, use props to define it (e.g. className="myCustomClassForTextSize")
+   */
+  iconSize?: CurrencySymbolSize;
+
+  /**
+   * If `true`, hovering over the `CurrencySymbol` will display the
+   * name of `currency` in a `Tooltip`. If `false` then the `CurrencySymbol`
+   * won't have any `Tooltip`.
+   *
+   * Defaults to `true`.
+   */
+  describeCurrencyInTooltip?: boolean;
+
+  /**
+   * The symbology to use when displaying the currency.
+   *
+   * Defaults to `CurrencySymbology.Symbol`.
+   */
+  symbology?: CurrencySymbology;
+}
+
+/**
+ *
+ * @param currency: Currency. The currency to get the Icon for (e.g. Currency.Eth)
+ * @param iconSize: CurrencySymbolSize. The Icon size (width and height), in pixels.
+ * @returns
+ */
+const getCurrencyIcon = ({
+  currency,
+  iconSize = CurrencySymbolSize.Small,
+  ...props
+}: {
+  iconSize: CurrencySymbolSize;
+  currency: Currency;
+}): JSX.Element => {
+  switch (currency) {
+    case Currency.Usd:
+      return <UsdSymbol {...props} />;
+    case Currency.Usdc:
+      return <UsdcSymbol {...props} width={iconSize} height={iconSize} />;
+    case Currency.Dai:
+      return <DaiSymbol {...props} width={iconSize} height={iconSize} />;
+    case Currency.Weth:
+      return <WethSymbol {...props} width={iconSize} height={iconSize} />;
+    case Currency.Eth:
+      return <EthSymbol {...props} width={iconSize} height={iconSize} />;
+    default:
+      // TODO: We haven't created symbols for `Currency.Gas` yet.
+      throw new Error(
+        `Error creating CurrencySymbol: unsupported Currency: "${currency}".`,
+      );
+  }
+};
+
+/**
+ * Returns a JSX.Element containing the currency's symbol, acronym or icon.
+ * @param currency: Currency. The currency to get the symbology for (e.g. Currency.Eth)
+ * @param symbology: CurrencySymbolSize. The size to use for Icon Symbology.
+ *                   For other symbologies it is ignored. We suggest you use props to define
+ *                   other symbologies sizes, as these are not SVGs but texts, instead. (e.g. className="myCustomClassForTextSize")
+ * @param symbology: CurrencySymbology. The symbology to use (e.g. CurrencySymbology.Symbol)
+ * @returns: JSX.Element. The currency's symbol, acronym or icon inside a JSX.Element where all extra props will be applied.
+ */
+const getCurrencySymbology = ({
+  currency,
+  iconSize = CurrencySymbolSize.Small,
+  symbology = CurrencySymbology.Symbol,
+  ...props
+}: {
+  currency: Currency;
+  iconSize?: CurrencySymbolSize;
+  symbology: CurrencySymbology;
```

Comment:
The `symbology` param is more logical to put after `currency` and not as the last param. That's because `iconSize` comes "downstream" of the `symbology` param.

Reply:
True!